packages/widget/src/controllers/wallet-manager/token-balance.ts

```diff
@@ -100,4 +119,29 @@ export class TokenBalanceController implements ReactiveController {
       this.host.requestUpdate();
     }.bind(this)();
   };
+
+  suscribeSubstrateBalanceUpdate = (resource: SubstrateResource): void => {
+    const { substrateProvider: apiPromise, signerAddress } = this.walletContext
+      .value?.substrateWallet as SubstrateWallet;
+
+    void async function (this: TokenBalanceController) {
+      try {
+        this.loadingBalance = true;
+        const tokenBalance = await getAssetBalance(
+          apiPromise as ApiPromise,
+          resource.assetID as number,
+          signerAddress as string
+        );
+        this.host.requestUpdate();
```

Comment:
this should go before `await getAssetBalance˙

Reply:
totally true